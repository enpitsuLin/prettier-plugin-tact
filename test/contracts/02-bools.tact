import "@stdlib/deploy";

contract Bools with Deployable {
 
    // contract persistent state variables
    b1: Bool = true;
    b2: Bool = false;
    b3: Bool;

    init() {
        self.b3 = !self.b2;
    }

    receive("show all") {
        dump(self.b1);
        dump(self.b2);
        dump(self.b3);
    }

    receive("show ops") {
        let b: Bool = true; // temporary variable
        dump(b);

        b = self.b1 && self.b2 || !self.b3;
        dump(b);

        dump(self.b1 == true);
        dump(self.b1 == self.b2);
        dump(self.b1 != self.b2);
    }

    get fun result(): Bool {
        return self.b1;
    }
}