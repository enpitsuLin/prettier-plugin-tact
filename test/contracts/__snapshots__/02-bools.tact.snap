import "@stdlib/deploy";

contract Bools with Deployable {
    // contract persistent state variables
    b1: Booltrue;b2: Boolfalse;b3: Bool;

    receive("show all") {
        dump(self.b1);dump(self.b2);dump(self.b3);
    }

    receive("show ops") {
        let b: Bool = true;
        // temporary variable
        dump(b);
        b = self.b1 && self.b2 || !self.b3;
        dump(b);
        dump(self.b1 == true);dump(self.b1 == self.b2);dump(self.b1 != self.b2);
    }

    get fun result(): Bool {
        return self.b1;
    }
}